### ISSUE TITLE
CA1829: Extend the rule to cover IReadOnlyCollection<T>.Count

### ISSUE THREAD

**Issue by Evangelink** [dotnet/roslyn-analyzers#2974]
#### Analyzer package

[Microsoft.CodeAnalysis.FxCopAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.FxCopAnalyzers)

#### Package Version

v2.9.5 (Latest)

#### Diagnostic ID

CA1829

#### Repro steps

```csharp
using System.Collections.Generic;
using System.Linq;

public class Foo
{
    public IReadOnlyCollection<int> GetData() => null;
    public int M() => GetData().Count();
}
```

#### Expected behavior

I expect to have a diagnostic raised against `GetData().Count()`.

#### Actual behavior

No diagnostic.
